packages/guides-restructured-text/src/RestructuredText/Directives/EpigraphDirective.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace phpDocumentor\Guides\RestructuredText\Directives;
+
+use phpDocumentor\Guides\Nodes\Node;
+use phpDocumentor\Guides\Nodes\QuoteNode;
+use phpDocumentor\Guides\RestructuredText\Nodes\CollectionNode;
+use phpDocumentor\Guides\RestructuredText\Parser\Directive;
+
+/**
+ * An epigraph is an apposite (suitable, apt, or pertinent) short inscription,
+ * often a quotation or poem, at the beginning of a document or section.
+ * The "epigraph" directive produces an "epigraph"-class block quote.
+ *
+ * https://docutils.sourceforge.io/docs/ref/rst/directives.html#epigraph
+ */
+class EpigraphDirective extends SubDirective
+{
+    public function getName(): string
+    {
+        return 'epigraph';
+    }
+
+    /** {@inheritDoc}
+     *
+     * @param Directive $directive
+     */
+    protected function processSub(
+        CollectionNode $collectionNode,
+        Directive $directive,
+    ): Node|null {
+        return new QuoteNode($collectionNode->getChildren(), ['epigraph']);
```

Comment:
Should we not add an extra class to the last paragraph node to mark it as attribution?

Reply:
The last paragraph is only an attribution if it starts with two -- so we have to add an extra test for that then

Reply:
Maybe an extra rule for that would make sense? that is only applied to this specific directive. As you might want to remove the `--` starter from the final node as well? 

Let's do that in a followup PR, because this directive needs more attention :-)